.github/workflows/main.yml

```diff
@@ -729,6 +734,35 @@ jobs:
       env:
         GH_TOKEN: ${{ github.token }}
 
+  # Test the `wasmtime-wasi-keyvalue` crate. Split out from the main tests
+  # because it needs additional database service.
+  test_wasi_keyvalue:
+    strategy:
+      matrix:
+        os: [ "ubuntu-latest", "macOS-latest" ]
+    name: Test wasi-keyvalue on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    needs: determine
+    if: needs.determine.outputs.run-wasi-keyvalue
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        submodules: true
+    - uses: ./.github/actions/install-rust
+    # Install Rust targets
+    - run: rustup target add wasm32-wasi
+    # Setup redis server
+    - uses: shogo82148/actions-setup-redis@v1
```

Comment:
Is there perhaps a more "official" action to use here to setup redis? We in theory try to avoid third-party dependencies on actions because of security concerns, but if it's a complicated and/or widely-enough-used action we can make an exception.

Reply:
I am now using GitHub's [Redis service containers](https://docs.github.com/en/actions/using-containerized-services/creating-redis-service-containers) to start a Redis service, but the drawback is that it can only run on Linux. Therefore, the previous macOS testing has been removed, but I think this is an acceptable compromise.

Reply:
Yeah that sounds reasonable to me too, thanks!